File: src/main/java/xyz/nulldev/ts/api/http/task/TaskStatusRoute.java
/*
 * Copyright 2016 Andy Bao
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package xyz.nulldev.ts.api.http.task;

import org.json.JSONObject;
import spark.Request;
import spark.Response;
import uy.kohesive.injekt.InjektKt;
import xyz.nulldev.ts.api.http.TachiWebRoute;
import xyz.nulldev.ts.api.task.Task;
import xyz.nulldev.ts.api.task.TaskManager;
import xyz.nulldev.ts.library.Library;
import xyz.nulldev.ts.util.LeniantParser;

/**
 * Project: TachiServer
 * Author: nulldev
 * Creation Date: 27/08/16
 */
public class TaskStatusRoute extends TachiWebRoute {

    public static final String KEY_COMPELTE = "complete";
    public static final String KEY_DETAILS = "details";

    private TaskManager taskManager = InjektKt.getInjekt().getInstance(TaskManager.class);

    @Override
    public Object handleReq(Request request, Response response) throws Exception {
        //Parse arguments
        Long taskId = LeniantParser.parseLong(request.params(":taskId"));
        if (taskId == null) {
            return error("TaskID must be specified!");
        }
        //Get task
        Task task = taskManager.getTask(taskId);
        if(task == null) {
            return error("No task found with the specified task ID!");
        }
        JSONObject result = success();
        result.put(KEY_COMPELTE, task.isComplete());
        result.put(KEY_DETAILS, task.getTaskStatus());
        return result;
    }
}

File: src/main/java/xyz/nulldev/ts/api/http/task/TaskStatusRoute.kt
/*
 * Copyright 2016 Andy Bao
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package xyz.nulldev.ts.api.http.task

import spark.Request
import spark.Response
import uy.kohesive.injekt.Injekt
import uy.kohesive.injekt.api.get
import xyz.nulldev.ts.api.http.TachiWebRoute
import xyz.nulldev.ts.api.task.TaskManager

/**
 * Project: TachiServer
 * Author: nulldev
 * Creation Date: 30/09/16
 */
class TaskStatusRoute : TachiWebRoute() {

    private val taskManager: TaskManager = Injekt.get()

    override fun handleReq(request: Request, response: Response): Any {
        //Parse arguments
        val taskId = request.params(":taskId")?.toLong()
                ?: return error("TaskID must be specified!")
        //Get task
        val task = taskManager.getTask(taskId)
                ?: return error("No task found with the specified task ID!")
        val result = success()
        result.put(KEY_COMPELTE, task.complete)
        result.put(KEY_DETAILS, task.taskStatus)
        return result
    }

    companion object {
        val KEY_COMPELTE = "complete"
        val KEY_DETAILS = "details"
    }
}